nixos/modules/services/misc/matrix-appservices/as-formats.nix

```diff
@@ -0,0 +1,187 @@
+{ systemConfig, asConfig, lib, pkgs, ... }:
+
+with lib;
+let
+  inherit (systemConfig.services.matrix-appservices)
+    homeserverURL
+    homeserverDomain;
+  package = asConfig.package;
+  pname = getName package;
+  command = "${package}/bin/${pname}";
+  getDefaultConfig = file: builtins.fromJSON (builtins.readFile
+    (pkgs.runCommand file { } ''
+      file=$(find ${package.src} -name '${file}' | head -1)
+      ${pkgs.yq}/bin/yq . $file > $out
+    '')
+  );
```

Comment:
Here is the IFD use, I want to get the default merging to happen on startup with yq and not do it in nix.

Reply:
just fixed this in the formats file